### ISSUE TITLE
[bug] cannot find -lbls_signatures in devtoolset-7 build env

### ISSUE THREAD

**Issue by taoshengshi** [filecoin-project/go-filecoin#3120]
**Describe the bug**
[root@localhost go-filecoin]# make all
go run ./build/*.go build
Building go-filecoin...
git log -n 1 --format=%H
go build -ldflags  -X github.com/filecoin-project/go-filecoin/flags.Commit=ec25290858c6dd61548528a68fd87ebc94ee63e0 -v -o go-filecoin .
github.com/filecoin-project/go-filecoin/bls-signatures
github.com/filecoin-project/go-filecoin/bls-signatures
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld: cannot find -lbls_signatures
collect2: error: ld returned 1 exit status
Command 'go build -ldflags  -X github.com/filecoin-project/go-filecoin/flags.Commit=ec25290858c6dd61548528a68fd87ebc94ee63e0 -v -o go-filecoin .' failed: exit status 2
exit status 1
make: *** [all] Error 1

**To Reproduce**

1. Install CentOS SCLo RH repository:
yum install centos-release-scl-rh   
2. Install llvm-toolset-7-clang-tools-extra rpm package:
yum install llvm-toolset-7-clang-tools-extra     
3. scl enable to add PATH       
scl enable devtoolset-7 bash
4. make deps
5. make all

**Expected behavior**
ld should be able to find the bls_signatures and make all should be successful.
I am confused with  how go build search the bls_signatures dependences and I can manually copy .a file to fix this: `cp bls-signatures/lib/libbls_signatures_ffi.a //usr/lib/libbls_signatures.a`.

**Version information**

- Go: go version go1.12.7 linux/amd64
- Rust: rustc 1.36.0 (a53f9df32 2019-07-03)
- Cargo: argo 1.36.0 (c4fcfb725 2019-05-15)
- Commit: ec25290858c6dd61548528a68fd87ebc94ee63e0
- Go Env: 
```go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/tshi/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/tshi/code/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build287286147=/tmp/go-build -gno-record-gcc-switches"
```
- CC: 
```
gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Additional context**
this is the master branch of go-filecoin
./go-filecoin version
commit: ec25290858c6dd61548528a68fd87ebc94ee63e0
[root@localhost go-filecoin]# git log --graph --pretty='%h %d %s' -n10
ec25290  (HEAD, origin/master, origin/HEAD, master) Remove BlockHeight from chain store (#3076)

**Comment by frrist** [comment]
The Makefile doesn't run the `deps` command:
```
[root@localhost go-filecoin]# make all
go run ./build/*.go build
Building go-filecoin...
```
Filecoin cannot be built without first running `deps`, i.e. `go run ./build/*go deps`. This is a bug in the Makefile, thanks for filing this.

**Comment by frrist** [comment]
@taoshengshi if you run `go run ./build/*go deps` and then run `go run ./build/*go build` it should fix the issue you are having, can you confirm here after you give that a try?

**Comment by frrist** [comment]
cc @sidke could you take a look at this?

**Comment by taoshengshi** [comment]
@frrist @sidke my team mate figure out the bug exists in the scripts of `scripts/install-bls-signatures.sh` :
```
find "${subm_dir}" -type f -name libbls_signatures_ffi.a -exec cp -- "{}" ./bls-signatures/lib/ \;
```
->
```
find "${subm_dir}" -type f -name libbls_signatures_ffi.a -exec cp -- "{}" ./bls-signatures/lib/libbls_signatures.a \;
```
it works successfully with above copy changes from libbls_signatures_ffi.a to libbls_signatures.a .

**Comment by frrist** [comment]
Great! Those changes we added in the fix #3123
